src/main/java/com/tictactoe/HumanPlayer.java

```diff
@@ -1,27 +1,42 @@
 package com.tictactoe;
 
 public class HumanPlayer {
-    private String symbol;
+    private String input;
     private UserInput receiver;
-    
-    public HumanPlayer(String symbol, UserInput receiver) {
-        this.symbol = symbol;
-        this.receiver = receiver;
+    private UserInputValidator validator = new UserInputValidator();
+    private String symbol;
+
+    public HumanPlayer(UserInput receiver, String symbol) {
+       this.receiver = receiver;
+       this.symbol = symbol;
+    }
+
+    public String getInput() {
+        this.input = this.receiver.obtainInput();
+        return this.input;
     }
 
     public String getSymbol() {
-        return symbol;
+        return this.symbol;
     }
 
-    public void makeMove(Board board) {
-        board.drawGridWithOnlyCellNumberId();
-        int cellSelection = receiver.obtainValidCellSelection();
-        int cellIndex = cellSelection - 1;
-        if (board.getSymbol(cellIndex).equals(" ")) {
-            board.insertSymbol(symbol, cellIndex);
-        } else {
-            System.out.println("Cell already occupied. Please select an empty cell.");
-            makeMove(board);
+    public int obtainValidCellSelection() {
+        int cell;
+        System.out.println("Enter your cell selection (1 - 9): ");
+        while (true)  {
```

Comment:
Using `while(true)` along with a break will certainly get the job done, but I prefer to use a specific terminating case with my while statements. For example, `bool player_has_no_selection = true; while(player_has_no_selection) { ...}`. This can help prevent a potential infinite loop